mlir/lib/Dialect/Tosa/Transforms/TosaProfileCompliance.cpp

```diff
@@ -464,9 +464,16 @@ LogicalResult TosaProfileCompliance::checkInvalid(Operation *op) {
   CheckCondition condition = CheckCondition::invalid;
   const auto maybeProfDef = getOperatorDefinition<Profile>(op, condition);
   const auto maybeExtDef = getOperatorDefinition<Extension>(op, condition);
+  if (failed(maybeProfDef) && failed(maybeExtDef))
+    return success();
+
+  bool hasEntry = false;
```

Comment:
How about :

```
bool hasEntry = (succeeded(maybeProfDef) && !maybeProfDef->empty()) ||
                  (succeeded(maybeExtDef) && !maybeExtDef->empty());
```

It keeps it short and readable while avoid value access and implicit conversion.